### PR TITLE
Fixed cast to smaller integer type on MSYS2 Clang64.

### PR DESCRIPTION
**Pull request by nmoinvaz**
```c
zlib/crc32_braid.c:128:24: error: cast to smaller integer type 'unsigned long'
 from 'const uint8_t *' (aka 'const unsigned char *')
```
See #1370.

### CODE REVIEW

**Comment by mtl1979** [comment]
This might mask another issue with that compiler. `z_size_t` should not be `unsigned long` for that compiler target.

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
This fixes the compiler warning for me, and all the tests pass. Can it be merged? CI failures seem unrelated.

@mtl1979 `zbuild.h` defines
```c++
#if defined(ZLIB_COMPAT)
...
#  define z_size_t unsigned long
#else
...
#  define z_size_t size_t
#endif
```
It makes me think this is needed for compatibility with zlib.

**Comment by mtl1979** [comment]
@dzenanz zlib didn't even support MSYS2... It had only minimal support for MinGW. For most other operating systems, `unsigned long` works correctly.

**Comment by dzenanz** [comment]
I have the same issue on Windows 11 with Visual Studio, see #1374.

**Comment by mtl1979** [comment]
@Dead2 will merge when he has time. My opinion that we really should try to avoid using `z_size_t` in anything else than function parameter types in `*.c` files doesn't block merging this PR.

**Pull request merged**
